File: scripts/add-prices.mjs
import { readFileSync, writeFileSync } from 'fs';

const file = readFileSync('/app/order-triage/page.tsx', 'utf8');
const lines = file.split('\n');

// Seed for reproducible but varied prices
const prices = [3.45, 7.20, 2.85, 12.50, 4.99, 6.75, 8.30, 1.95, 5.60, 9.15, 
                11.20, 3.80, 7.45, 2.50, 14.99, 6.10, 4.25, 8.90, 3.15, 10.50,
                5.75, 7.80, 2.30, 9.60, 4.40, 11.85, 3.50, 6.95, 8.20, 1.75,
                13.40, 5.25, 7.10, 2.90, 10.15, 4.60, 9.35, 6.50, 3.70, 8.55,
                12.00, 5.90, 7.35, 2.65, 11.50, 4.80, 9.75, 6.20, 3.95, 8.10,
                14.25, 5.40, 7.65, 2.15, 10.80, 4.15, 9.50, 6.85, 3.30, 8.70,
                12.60, 5.05];

let priceIdx = 0;
const result = [];

for (let i = 0; i < lines.length; i++) {
  result.push(lines[i]);
  
  // Match "pickedQty: N," in sample data (not in type definition)
  const match = lines[i].match(/^(\s+)pickedQty: (\d+),$/);
  if (match) {
    const indent = match[1];
    const pickedQty = parseInt(match[2]);
    
    // Check we're in sample data (next line shouldn't be a type field)
    // and that we haven't already added unitPrice
    const nextLine = lines[i + 1] || '';
    if (!nextLine.includes('unitPrice')) {
      // Look back to find quantity
      let qty = pickedQty;
      for (let j = i - 1; j >= Math.max(0, i - 5); j--) {
        const qtyMatch = lines[j].match(/quantity: (\d+)/);
        if (qtyMatch) {
          qty = parseInt(qtyMatch[1]);
          break;
        }
      }
      
      const unitPrice = prices[priceIdx % prices.length];
      const lineTotal = parseFloat((qty * unitPrice).toFixed(2));
      priceIdx++;
      
      result.push(`${indent}unitPrice: ${unitPrice},`);
      result.push(`${indent}lineTotal: ${lineTotal},`);
    }
  }
}

writeFileSync('/app/order-triage/page.tsx', result.join('\n'));
console.log(`Added unitPrice and lineTotal to ${priceIdx} line items`);

File: scripts/add-prices.py
import re
import random

file_path = '/app/order-triage/page.tsx'

with open(file_path, 'r') as f:
    content = f.read()

lines = content.split('\n')
new_lines = []
i = 0

# Price ranges for realistic data
prices = [1.25, 1.50, 1.75, 2.00, 2.25, 2.50, 2.75, 3.00, 3.25, 3.50, 3.75, 4.00, 4.25, 4.50, 4.99, 5.25, 5.50, 5.99, 6.25, 6.50, 7.25, 7.99, 8.50, 9.99, 10.50, 12.00, 14.99, 15.50]
price_idx = 0

while i < len(lines):
    line = lines[i]
    new_lines.append(line)
    
    # Match pickedQty lines in lineItems data (not in type definitions)
    match = re.match(r'^(\s+)pickedQty:\s*(\d+),\s*$', line)
    if match:
        indent = match.group(1)
        picked = int(match.group(2))
        
        # Check next line doesn't already have unitPrice
        if i + 1 < len(lines) and 'unitPrice' not in lines[i + 1]:
            # Also check we're in a data block (not type def) by checking for quantity line above
            # Look back for quantity
            found_quantity = False
            for back in range(1, 5):
                if i - back >= 0 and 'quantity:' in lines[i - back]:
                    # Extract quantity value
                    q_match = re.search(r'quantity:\s*(\d+)', lines[i - back])
                    if q_match:
                        qty = int(q_match.group(1))
                        found_quantity = True
                        break
            
            if found_quantity:
                unit_price = prices[price_idx % len(prices)]
                price_idx += 1
                line_total = round(unit_price * qty, 2)
                new_lines.append(f'{indent}unitPrice: {unit_price},')
                new_lines.append(f'{indent}lineTotal: {line_total},')
    
    i += 1

with open(file_path, 'w') as f:
    f.write('\n'.join(new_lines))

print(f"Done! Added unitPrice and lineTotal to {price_idx} line items.")
